.github/workflows/pythonapp.yml

```diff
@@ -31,7 +31,7 @@ jobs:
       shell: bash
       run: |
         python -m pip install --upgrade pip
-        pip install -r requirements.txt $EXTRAPKGS
+        pip install -r requirements.txt $EXTRAPKGS rstcheck
```

Comment:
If we put `rstcheck` into `requirements.txt`, that'd eliminate the duplicate definition of the module.

Reply:
I thought `requirements.txt` was what was required to install the module, not necessarily test it. And I thought `setup_requires` was what was needed to build the module, not install it... Basically I want to require this for testing but not for an end user. How can I do that?

Reply:
> And I thought `setup_requires` was what was needed to build the module, not install it

In the sense of compilation, yes, `setup_requires` is what's needed to build the package.  An alternate way to think about this is that `setup_requires` is what's needed to execute `setup.py`.  For taskgraph, I'm pretty sure the only thing required is `setuptools_scm`.

> Basically I want to require this for testing but not for an end user. How can I do that?

Well, that test file won't import unless `rstcheck` is installed!  So in a sense, it's already required.

There's less consensus on how to do this, but there's also less consensus on how to run tests.  I suggest keeping a second `requirements.txt` file for dev/test dependencies ([InVEST does this](https://github.com/natcap/invest/blob/main/requirements-dev.txt)).  But, if you really really want to have your test runner _also_ install test dependencies, then take a look at [`python setup.py test`](https://stackoverflow.com/q/36391135/299084).

Reply:
I agree w/ you about being required inherently for pytest. I think what I'd like to do then is remove `rstcheck` from `setup.py` but leave it in `pythonapp.yml` would that be okay if I did it that way?

Reply:
Yep, I think that'd be just fine.

Reply:
okay i did it in 9526ab4b993245ddfd40a1cd6fdfabb5e6e12e06